app/scripts/modules/core/src/pipeline/details/SingleExecutionDetails.tsx

```diff
@@ -38,108 +30,155 @@ export interface ISingleExecutionRouterStateChange extends IStateChange {
+  useEffect(() => {
+    const subscription = someActive && scheduler.subscribe(() => refreshExecution());
+
+    return () => {
+      subscription && subscription.unsubscribe();
+    };
+  }, [someActive]);
+
+  // Responsible for listening to state changes and updating executionId
+  useEffect(() => {
+    const subscription = stateEvents.stateChangeSuccess.subscribe((stateChange: ISingleExecutionRouterStateChange) => {
+      if (
+        !stateChange.to.name.includes('pipelineConfig') &&
+        !stateChange.to.name.includes('executions') &&
+        (stateChange.toParams.application !== stateChange.fromParams.application ||
+          stateChange.toParams.executionId !== stateChange.fromParams.executionId)
+      ) {
+        setExecutionId(stateChange.toParams.executionId);
+        const lineage = traverseLineage(execution);
+        if (lineage.includes(stateChange.toParams.executionId)) {
+          setTransitioningToAncestor(stateChange.toParams.executionId);
         }
-      },
-    );
-    this.getExecution();
-  }
+      }
+    });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [execution]);
 
-  public componentWillUnmount(): void {
-    if (this.executionScheduler) {
-      this.executionScheduler.unsubscribe();
-    }
-    if (this.executionLoader) {
-      this.executionLoader.unsubscribe();
+  // Responsible for clearing transitioningToAncestor (allow rendering main execution again)
+  useEffect(() => {
+    if (transitioningToAncestor === execution?.id) {
+      setTransitioningToAncestor('');
     }
-    this.stateChangeSuccessSubscription.unsubscribe();
-  }
+  }, [transitioningToAncestor, execution?.id]);
 
-  private showDurationsChanged = (event: React.ChangeEvent<HTMLInputElement>): void => {
+  // Responsible for propagating showDurations back to the filterModel
+  React.useEffect(() => {
+    ExecutionState.filterModel.asFilterModel.sortFilter.showDurations = showDurations;
+  }, [showDurations]);
+
+  const { result: pipelineConfigs } = useLatestPromise<IPipeline[]>(() => {
+    app.pipelineConfigs.activate();
+    return app.pipelineConfigs.ready();
+  }, []);
+  const pipelineConfig =
+    pipelineConfigs && execution && pipelineConfigs.find((p: IPipeline) => p.id === execution.pipelineConfigId);
+
+  // Responsible for propagating showDurations changes to the filterModel
+  useEffect(() => {
+    ExecutionState.filterModel.asFilterModel.sortFilter.showDurations = showDurations;
+  }, [showDurations]);
+
+  const showDurationsChanged = (event: React.ChangeEvent<HTMLInputElement>): void => {
     const checked = event.target.checked;
-    // TODO: Since we treat sortFilter like a store, we can force the setState for now
-    //       but we should eventually convert all the sortFilters to be a valid redux
-    //       (or similar) store.
-    this.state.sortFilter.showDurations = checked;
-    this.setState({ sortFilter: this.state.sortFilter });
+    setShowDurations(checked);
```

Comment:
Totally. I initially did a direct port of the old implementation which both read from and wrote to the sort filter and it was painfully slow (I think because it hits `localStorage`), so I went to the other extreme of separating the toggle state and the sync code, but that isn't necessary. Thanks.